### ISSUE TITLE
key phrase extractions

### ISSUE THREAD

**Issue by prabirshrestha** [guillaume-be/rust-bert#45]
I filled a previous issue on this but seems to not be as expected. https://github.com/guillaume-be/rust-bert/issues/40. `text` seems to be either `POSITIVE` or `NEGATIVE` instead of actual labels. It seems to be doing `Sentiment analysis` instead of generating tags/labels.

```
[[Label { text: "POSITIVE", score: 0.9605776071548462, id: 1, sentence: 0 }], [Label { text: "NEGATIVE", score: 0.7535297274589539, id: 0, sentence: 1 }, Label { text: "POSITIVE", score: 0.2753136157989502, id: 1, sentence: 1 }], [Label { text: "POSITIVE", score: 0.9856524467468262, id: 1, sentence: 2 }]]
```

I have bunch of data and would like to extract key phrases similar to fast text multi label classification. https://fasttext.cc/docs/en/supervised-tutorial.html#multi-label-classification.

here is an example:

```
Which baking dish is best to bake a banana bread ?
```

should return something like

```
__label__baking 1.00000 __label__bananas 0.939923 __label__bread 0.592677
```

For this case the key phrases was `baking`, `bananas` and `bread`.

**Comment by guillaume-be** [comment]
@prabirshrestha Apologies if I closed the previous issue too early. 
Do you have a trained model available with one of the architectures supported by the crate to extract the labels you are trying to predict?

The target labels are provided in the `config.json` file of the model:
```json
"id2label": {
  "0": "NEGATIVE",
  "1": "POSITIVE"
},
```

For custom classes, you could store the class mapping in the config as: 
```json
"id2label": {
  "0": "baking",
  "1": "banana",
  "2": "bread",
  "3": "pineapple"
},
```

You would however need a model that was trained to predict the output classes you are targeting and that would map the output to the right output space dimension. This model can be trained in this library if you implement your training loop in Rust, or in Python (with several examples of multi-label transformers training available, for example: https://github.com/ThilinaRajapakse/simpletransformers#minimal-start-for-multilabel-classification). 

With these model weights available, you need to use the `sequence_classification` pipeline (of which the `sentiment` pipeline is a special case.

**Comment by prabirshrestha** [comment]
No worries on closing the issue. Even I was thinking if I should had commented it there before opening here.

I would personally love the training to also be done in rust primarily because I want to automate everything with rust.

I don't have a trained model. ML is a new topic for me so still learning.

**Comment by guillaume-be** [comment]
Hello,

Training the models provided in the library is entirely feasible, they include the features used for fine-tuning including dropout. I am planning to include some training utilities, but not in the near future as there are a number of higher priority items I'd like to address first.

The `tch-rs` library contains a few examples on how to train these models in Rust: https://github.com/LaurentMazare/tch-rs/tree/master/examples/mnist.

Note however that the finetuning of the language models in this library usually requires a few additional tricks to be effective. If you are starting your journey in machine learning, I'd still highly recommend getting started with Python as a very large amount of resources are available to learn about the topic.
